WcaOnRails/app/mailers/competitions_mailer.rb

```diff
@@ -67,9 +67,9 @@ def submit_report_nag(competition)
     )
   end
 
-  def results_submitted(competition, message, file_path)
+  def results_submitted(competition, message, file_name, file)
     @message = message
-    attachments[File.basename(file_path)] = File.read(file_path)
+    attachments[file_name] = File.read(file)
```

Comment:
Honestly, I don't see any value in using the filename that came from the user. I think it would be nice for the WRT if we generate a filename here ourselves, maybe even including a timestamp just in case people submit multiple times for the same competition. I'm thinking something like `file_name = "Results for #{competition.id} {Time.now.utc}.json"` or something.

Reply:
Also, could we change this method to take the contents of the file, rather than a filename? I think that's a cleaner abstraction than "pass me a file", and should address the issue you ran into while trying to mock this method.

Reply:
Well it's just that it could make a reallllllly long parameter and I'm not sure how that works. But if you're confident that it scales, I'm happy making the change.
I like the idea of generating the file name too!

Reply:
I think <competitionId>.json would make sense (and is also the most simply naming scheme I can think of).

Reply:
If we use ISO8601, we can generate the file name like this
`"Results_#{competition.id}_#{Time.now.utc.iso8601}.json"`
so the attachment file name would be
`Results_MyComp152017_2018-02-24T05:44:05Z.json`
and it could be saved to disk as
`Results_MyComp152017_2018-02-24T05_44_05Z.json`

I made the other change and we are now passing the file contents.